1-integrity/CoreForeignKeys.pl

```diff
@@ -61,135 +68,35 @@ =head1 DESCRIPTION
-    col1 => 'xref_id',
-    table2 => 'xref',
-    col2 => 'xref_id',
-    both_ways => 0,
-);
-
-$test_result &= DBUtils::CheckForOrphans::check_orphans(
-    helper => $helper,
-    logger => $log,
-    table1 => 'xref',
-    col1 => 'external_db_id',
-    table2 => 'external_db',
-    col2 => 'external_db_id',
-    both_ways => 0,
-);
-
-$test_result &= DBUtils::CheckForOrphans::check_orphans(
-    helper => $helper,
-    logger => $log,
-    table1 => 'dna',
-    col1 => 'seq_region_id',
-    table2 => 'seq_region',
-    col2 => 'seq_region_id',
-    both_ways => 0,
-);
-
-$test_result &= DBUtils::CheckForOrphans::check_orphans(
-    helper => $helper,
-    logger => $log,
-    table1 => 'seq_region',
-    col1 => 'coord_system_id',
-    table2 => 'coord_system',
-    col2 => 'coord_system_id',
-    both_ways => 0,
-);
-
-$test_result &= DBUtils::CheckForOrphans::check_orphans(
-    helper => $helper,
-    logger => $log,
-    table1 => 'assembly',
-    col1 => 'cmp_seq_region_id',
-    table2 => 'seq_region',
-    col2 => 'seq_region_id',
-    both_ways => 0,
-);
-
-$test_result &= DBUtils::CheckForOrphans::check_orphans(
-    helper => $helper,
-    logger => $log,
-    table1 => 'marker_feature',
-    col1 => 'marker_id',
-    table2 => 'marker',
-    col2 => 'marker_id',
-    both_ways => 0,
-);
-
-$test_result &= DBUtils::CheckForOrphans::check_orphans(
-    helper => $helper,
-    logger => $log,
-    table1 => 'seq_region_attrib',
-    col1 => 'seq_region_id',
-    table2 => 'seq_region',
-    col2 => 'seq_region_id',
-    both_ways => 0,
-);
-
-$test_result &= DBUtils::CheckForOrphans::check_orphans(
-    helper => $helper,
-    logger => $log,
-    table1 => 'seq_region_attrib',
-    col1 => 'attrib_type_id',
-    table2 => 'attrib_type',
-    col2 => 'attrib_type_id',
-    both_ways => 0,
-);
-
-$test_result &= DBUtils::CheckForOrphans::check_orphans(
-    helper => $helper,
-    logger => $log,
-    table1 => 'misc_feature_misc_set',
-    col1 => 'misc_feature_id',
-    table2 => 'misc_feature',
-    col2 => 'misc_feature_id',
-    both_ways => 0,
-);
-
-$test_result &= DBUtils::CheckForOrphans::check_orphans(
-    helper => $helper,
-    logger => $log,
-    table1 => 'misc_feature_misc_set',
-    col1 => 'misc_set_id',
-    table2 => 'misc_set',
-    col2 => 'misc_set_id',
-    both_ways => 0,
-);
+
+my %tables_hash;
+
+if($filter_tables){
+    use Input::FilteredCoreForeignKeys;
+    %tables_hash = %$Input::FilteredCoreForeignKeys::core_foreign_keys;
+}
+else{
+    use Input::CoreForeignKeys;
+    %tables_hash = %$Input::CoreForeignKeys::core_foreign_keys;
+}
+
+for my $table (keys %tables_hash ) {
+
+    for my $test (keys %{ $tables_hash{$table} } ){
+
+        $test_result &= DBUtils::CheckForOrphans::check_orphans(
+            helper => $helper,
+            logger => $log,
+            table1 => $table,
+            col1 => $tables_hash{$table}{$test}{'col1'},
+            table2 => $tables_hash{$table}{$test}{'table2'},
```

Comment:
As a quick comment if you want to reduce the hash access a little bit you can pull the test hash description out before you run the `check_orphans()` code. There's two possible ways to do this. One is to re-assign the values in the hash in the next call like so:

```perl
my $test_def = $tables_hash{$table}{$test};
$test_result &= DBUtils::CheckForOrphans::check_orphans(
  helper => $helper,
  log => $log,
  table => $table, 
  col1 => $test_def{'col1'},
  # ... and so on
);
```

The other way to do this is to take advantage that your naming conventions are consistent between the test definition and the `check_orphans()` call e.g.

```perl
my $test_def = $tables_hash{$table}{$test};
$test_result &= DBUtils::CheckForOrphans::check_orphans(
  helper => $helper,
  log => $log,
  table => $table,
  %{$test_def} # you de-reference the hash $test_def and let Perl take these in as subroutine parameters
);
```

Both are good ways of reducing your lines of code.